src/models/machine_learning/MLmodel.jl

```diff
@@ -47,11 +48,13 @@ Feed-forward neural network.
 mutable struct NeuralNetwork{
     ChainType <: Lux.Chain,
     ComponentVectorType <: ComponentVector,
-    NamedTupleType <: NamedTuple
+    NamedTupleType <: NamedTuple,
+    TAR <: AbstractTarget
 } <: MLmodel
     architecture::ChainType
     θ::ComponentVectorType
     st::NamedTupleType
+    target::TAR
```

Comment:
We will have to think where to exactly place the `Target`, since the following #302 the main idea was to drop `MLmodel` altogether, and only have iceflow and mass balance models.

Reply:
Would this means just to move ML model inside a broader model? Or do you mean to get rid of Target? Notice that target has inside all the required internals to compute the VJPs and gradient, this information needs to be provided somewhere.

Reply:
As discussed yesterday, the `MLmodel` which goes inside `Model` would go, and `Target` would be placed inside `Law` or merged with it.